libraries/GCS_MAVLink/MAVLink_routing.cpp

```diff
@@ -140,6 +140,13 @@ bool MAVLink_routing::check_and_forward(GCS_MAVLINK &in_link, const mavlink_mess
     int16_t target_component = -1;
     get_targets(msg, target_system, target_component);
 
+    // handle messages from a mavlink receiver as if it had target_system = our system, target_component = 0
+    if ((msg.compid == MAV_COMP_ID_TELEMETRY_RADIO) &&
```

Comment:
does the msg not have  target_system?

Reply:
no. Pl see the post on this below.

Reply:
This precludes having redundant receivers on-board.

You can target the message easily enough - we use the same technique on things like ADSB receivers.  If no explicit target has been specified on the device then it waits for a heartbeat to arrive and (after filtering for appropriate vehicle type and `MAV_TYPE`).  It then locks onto the flight controller, targetting it and using a source system the same as that flight controller.  component ID can default to TELEMETRY_RADIO but should also be configurable (and that ID should have no relevance to the way the packet is processed on the flight controller, to allow for redundancy)